code/controllers/subsystem/jobs.dm

```diff
@@ -726,3 +726,135 @@ SUBSYSTEM_DEF(jobs)
 		new_id_change_records["[id_change_counter]"] = thisrecord
 		id_change_counter++
 	id_change_records = new_id_change_records
+
+// This proc will update all players EXP at once. It will calculate amount of time to add dynamically based on the SS fire time.
+/datum/controller/subsystem/proc/batch_update_player_exp(announce = FALSE)
+	// Right off the bat
+	var/start_time = start_watch()
+	// First calculate minutes
+	var/divider = 10 // By default, 10 deciseconds in 1 second
+	if(flags & SS_TICKER)
+		divider = 20 // If this SS ever gets made into a ticker SS, account for that
+
+	var/minutes = (wait / divider) / 60 // Calculate minutes based on the SS wait time (How often this proc fires)
```

Comment:
Can these not use the `SECONDS`/`MINUTES` defines?

Reply:
Those are `*10` and `*60` respectively. This needs to be hardcoded this way because `MINUTES` includes the `*10` on seconds itself, and that breaks this system